### DISCUSSION TITLE
Date picker format - DD.MM.YYYY

### DISCUSSION THREAD

**Discussion by karthickmerck** [emdgroup-liquid/liquid#279] [Q&A] [answered]
Please let us know how can we change the date format of the date picker to DD.MM.YYYY.  

Currently we are facing issues like by default it's system time zone. 
<img width="745" alt="img date format" src="https://user-images.githubusercontent.com/98765079/151838031-a264e530-5169-4e39-abbd-3d9219b4ba26.PNG">

**Comment by borisdiakur** [comment, marked as answer]
This is currently not possible with the `ld-input` component, because it simply acts as a wrapper around the native input field, fully relying on its built-in features, only enhancing it visually. I guess the reason for why the native input field does not offer this feature is because it doesn't make much sense to change the format at all, since (in most cases) the user has configured her or his system and browser in a way that dates are displayed in the user's preferred format, which the browser picks up and uses for the native input field. In other words, the screenshot above shows exactly the expected behaviour.

We could offer a separate web component, if the need justifies the effort. However, I'm curious to what would be the reason to change the format at all? I can think of reasons such as, the browser picking up the system format instead of the region format configured in the browser settings. But this is not the case with Chrome. Or you have a page which has `lang="de-DE"` so you expect it to display dates in `DD.MM.YYYY` format. But does the user expect this as well? Maybe the user is from the US? Maybe the user is from Spain, but prefers the US format?

The advantages of a native date input field are, that it is lightweight, respects user preferences, separates presentation format from wire format and is accessible out of the box (especially on mobile browsers) by offering the user a familiar UI with a logic she or he is accustomed to. Please take all this into account when considering a custom date picker.

Related question on SO: https://stackoverflow.com/questions/7372038/is-there-any-way-to-change-input-type-date-format
